import/index_java/src/org/vufind/index/FormatCalculator.java

```diff
@@ -209,6 +217,18 @@ protected String getFormatFromBibLevel(Record record, char bibLevel, char format
             // Monograph
             case 'm':
                 if (couldBeBook) {
+                    if (marc008 != null) {
+                        // Check 008/23 Form of item
+                        switch (marc008.getData().toLowerCase().charAt(23)) {
```

Comment:
Should we check for a particular length of 008 in addition to checking that it is not null, so that malformed/incomplete 008's don't cause errors?

Reply:
sounds good. I'm not sure if there is a standard length 008, but can check. Are you suggesting checking each time that there are at least the number of chars needed, e.g. 24 in this case but more elsewhere?